### PR TITLE
add display property to Prefix model

### PR DESCRIPTION
**Pull request by brandomando**
# Closes #[6189](https://github.com/nautobot/nautobot/issues/6189)
# What's Changed

    - add `display` property to `Prefix` model so that it displays as `<prefix>: <namespace.name>`
# Screenshots

### Before:
Screenshot 2024-09-04 at 2 18 42 PM

### After:
Screenshot 2024-09-04 at 2 21 16 PM

# TODO

- [x] Explanation of Change(s)

- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by HanlinMiao — APPROVED** [review]
Need a changelog but everything else looks good to me!

File: nautobot/ipam/tests/test_api.py [hunk 1]
```diff
@@ -443,6 +443,25 @@ def test_list_available_prefixes(self):
...
+            .filter(prefix_length__lt=128)
+            .exclude(type=choices.PrefixTypeChoices.TYPE_CONTAINER)
+            .first()
+        )
+        if prefix is None:
+            self.fail("Suitable prefix fixture not found")
+        url = reverse("ipam-api:prefix-list")
+        self.add_permissions("ipam.view_prefix")
+
+        response = self.client.get(url, **self.header, QUERY_STRING="depth=1")
```
Comment (glennmatthews) [inline nautobot/ipam/tests/test_api.py:461]: I'm not familiar with the use of `QUERY_STRING` as a header parameter, does it actually do something here? I'd have expected a more usual pattern to be `url = f"{url}?depth=1"` instead.

File: nautobot/ipam/tests/test_api.py [hunk 2]
```diff
@@ -443,6 +443,25 @@ def test_list_available_prefixes(self):
...
+        Test that the `display` field is correctly populated.
+        """
+        prefix = (
+            Prefix.objects.filter(ip_version=6)
+            .filter(prefix_length__lt=128)
+            .exclude(type=choices.PrefixTypeChoices.TYPE_CONTAINER)
+            .first()
+        )
+        if prefix is None:
+            self.fail("Suitable prefix fixture not found")
```
Comment (glennmatthews) [inline nautobot/ipam/tests/test_api.py:457]: What's the purpose of this? You're listing all prefixes below, what is this specific `prefix` used for?

**Comment by brandomando** [comment]
I submitted another commit to fix the ruff formatting and also the changelog fragment

File: nautobot/ipam/tests/test_api.py (hunk 1 above)
Comment (brandomando) [reply nautobot/ipam/tests/test_api.py:461]: Hi Glenn, this was used due to the test client that's being employed (Django's `test.client.RequestFactory`), which traces back to [here](https://github.com/django/django/blob/main/django/test/client.py#L399).  If you'd prefer that I use the `url` string instead, I can refactor.

File: nautobot/ipam/tests/test_api.py (hunk 2 above)
Comment (brandomando) [reply nautobot/ipam/tests/test_api.py:457]: Copy-pasta error. I will remove it. 🫢

File: nautobot/ipam/tests/test_api.py (hunk 1 above)
Comment (brandomando) [reply nautobot/ipam/tests/test_api.py]: Changed in [7cd2afd](https://github.com/nautobot/nautobot/pull/6190/commits/7cd2afd808a1ff2d5fdeb6a728954a31560a513e)

**Review by HanlinMiao — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**